Add modular support to repomanage (RhBug:1804720)

kontura opened the pull request:
In order to support modular content we have to load repodata with
repodata/modules.yaml present.
If repodata are not present in the target path there is no way to tell
which packages belong to which stream and repomanage will continue to
show packages from different streams as obsoleting each other.

Tests: https://github.com/rpm-software-management/ci-dnf-stack/pull/825

https://bugzilla.redhat.com/show_bug.cgi?id=1804720

pkratoch commented:
Also, unit tests must be adjusted to the new code. Currently, they are failing because "AttributeError: 'BaseStub' object has no attribute '_add_repo_to_sack'".

kontura commented on plugins/repomanage.py:
```diff
@@ -88,35 +112,68 @@ def run(self):
+                if len(sorted_stream_versions) < keepnum:
+                    new_sorted_stream_versions = sorted_stream_versions
+                else:
+                    new_sorted_stream_versions = sorted_stream_versions[-keepnum:]
```
nice, thanks!

kontura commented on doc/repomanage.rst:
```diff
@@ -33,6 +33,8 @@ Description
 
 `repomanage` prints newest or oldest packages in specified directory for easy piping to xargs or similar programs.
```
I have updated the docs.

kontura commented:
I have pushed an update, splitting the whole thing into couple more commits and added one for the unit tests. 
I have also moved around the logic for preparing modular data a bit, now it is only executed if the repo is fully loaded.

kontura commented:
Pushed additional commit which fixes a bug with duplicate packages (Discovered when adding a requested test for it).

pkratoch commented:
Thanks for the changes. The tests pass locally, so I am merging this.
